src/platform/utilities/feature-toggles/useFeatureToggle.js

```diff
@@ -1,27 +1,110 @@
+
+import { setData } from '../../forms-system/src/js/actions';
 import TOGGLE_NAMES from './featureFlagNames';
 
 export const toggleValuesSelector = state => state.featureToggles || {};
 
 export const loadingTogglesSelector = state => state?.featureToggles?.loading;
 
-export const useFeatureToggle = () => {
-  const useToggleValue = toggleName => {
-    if (!Object.values(TOGGLE_NAMES).includes(toggleName)) {
-      throw new Error(
-        `Invalid toggle name "${toggleName}". Did you add the toggle name to featureFlagNames.js?`,
-      );
-    }
-    return useSelector(state => toggleValuesSelector(state)?.[toggleName]);
-  };
+/**
+ * Configuration type for a single feature toggle
+ * @typedef { Object } ToggleConfig
+ * @property { string } toggleName - CamelCased name of the feature toggle example: 'profileUseExperimental'
+ * @property { string } [formKey] - Optional custom key for form / session storage
+ */
 
-  const useToggleLoadingValue = () => {
-    return useSelector(state => loadingTogglesSelector(state));
-  };
+/**
+ * Get feature toggle loading state
+ * @returns {Boolean} - feature flag loading state
+ */
+export const useToggleLoadingValue = () => {
+  return useSelector(state => loadingTogglesSelector(state));
+};
+
+/**
+ * Hook to sync feature toggles with form data and session storage
+ * @param {(string|ToggleConfig)[]} toggles - Array of toggle name strings or ToggleConfig objects
+ * @param {Function} [setStorageItem] - Storage setter function (defaults to sessionStorage.setItem)
+ */
+export const useFormFeatureToggleSync = (
+  toggles,
+  setStorageItem = window.sessionStorage.setItem.bind(window.sessionStorage),
+) => {
+  const dispatch = useDispatch();
+  const formData = useSelector(state => state?.form?.data ?? {});
+  const isLoadingFeatureFlags = useToggleLoadingValue();
+  const featureToggles = useSelector(toggleValuesSelector);
+
+  // Normalize toggle configurations
+  const normalizedToggles = useMemo(
+    () =>
+      toggles.map(toggle => {
+        if (typeof toggle === 'string') {
+          return { toggleName: toggle, formKey: toggle };
+        }
+        return { ...toggle, formKey: toggle.formKey ?? toggle.toggleName };
+      }),
+    [toggles],
+  );
+
+  // Calculate updates needed for form data and storage
+  const updates = useMemo(
+    () => {
+      if (isLoadingFeatureFlags) return {};
 
+      return normalizedToggles.reduce((acc, { toggleName, formKey }) => {
+        const toggleValue = featureToggles[(TOGGLE_NAMES?.[toggleName])];
+        // Only include in updates if the value differs from current form data
+        if (toggleValue !== formData?.[formKey]) {
+          setStorageItem(formKey, String(toggleValue));
+          return { ...acc, [formKey]: toggleValue };
+        }
+
+        return acc;
+      }, {});
+    },
+    [
+      normalizedToggles,
+      featureToggles,
+      isLoadingFeatureFlags,
+      formData,
+      setStorageItem,
+    ],
+  );
+
```

Comment:
Memoizing here so that the useEffect only fires when there are actual changes and helps prevent excessive re-render cycles.